From 1a336ad1a5f234d138e5f390d767f51f81760f51 Mon Sep 17 00:00:00 2001
From: Ashutosh Rawat <ashurawa@qti.qualcomm.com>
Date: Mon, 23 Feb 2026 15:47:08 +0530
Subject: [PATCH 16/40] qcacld-3.0: Fix NULL VDEV dereference

Potential NULL VDEV dereference

Add NULL check for VDEV

CRs-Fixed: 4436870
Change-Id: If6fc6404677e3cd04496fe071b31a5de0c8a3bb8
---
 core/hdd/src/wlan_hdd_green_ap.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/core/hdd/src/wlan_hdd_green_ap.c b/core/hdd/src/wlan_hdd_green_ap.c
index a571a8bd8c..f5be21b22e 100644
--- a/core/hdd/src/wlan_hdd_green_ap.c
+++ b/core/hdd/src/wlan_hdd_green_ap.c
@@ -359,6 +359,11 @@ QDF_STATUS wlan_hdd_send_green_ap_ll_ps_event(
 	struct vdev_osif_priv *osif_priv;
 	struct sk_buff *skb;
 
+	if (!vdev) {
+		hdd_err("VDEV is NULL");
+		return QDF_STATUS_E_FAILURE;
+	}
+
 	osif_priv = wlan_vdev_get_ospriv(vdev);
 	if (!osif_priv) {
 		hdd_err("OSIF priv is NULL");
-- 
2.34.1

